Fixed image caption select behaviour.

jorgefilipecosta opened the pull request:
A regression happened and now the caption of the image is automatically selected and the placeholder is never shown.
This is not intuitive for the user because the caption is focused but the cursor does not appear.

## How Has This Been Tested?
Add an image block with an image, click the image, verify you see the caption placeholder.
Write a caption click the image and see that the caption field lost the focus

## Screenshots (jpeg or gifs if applicable):
![feb-12-2018 22-20-29](https://user-images.githubusercontent.com/11271197/36123145-b5574fe8-1043-11e8-96d3-3984843f72c1.gif)
![feb-12-2018 22-16-19](https://user-images.githubusercontent.com/11271197/36123163-bfa361c6-1043-11e8-81c5-824458fc85b5.gif)

jorgefilipecosta committed:
Fixed image caption select behaviour.

A regression happened and now the caption of the image is automatically selected and the placeholder is never shown.

youknowriad commented:
I don't know if it's related to this PR, but I'm seeing a weird behavior related to the placeholder in this branch.

<img width="663" alt="screen shot 2018-02-13 at 13 24 43" src="https://user-images.githubusercontent.com/272444/36149776-561096e2-10c1-11e8-9f5d-9fd9701baa05.png">

See the white space above the placeholder in the screenshot, if I click the whitespace, the caption is focused properly but if I click the placeholder, the caption is not focused at all.

jorgefilipecosta commented:
Hi @youknowriad, that is a complex case. I tried to click the image many times until I was able to replicate it, and the whitespace disappeared as soon as I did an action like moved the mouse. I added a CSS rule that I think should fix the problem, could you confirm if it also fixed the problem in your case?
Thank you a lot for your review.

youknowriad commented:
@jorgefilipecosta It looks like a generic placeholder bug and in that case, I'd prefer if we could fix globally. If you think it's something unrelated to this PR, I'm fine dropping the temporary fix and moving forward with this PR and address the placeholder issue separately

jorgefilipecosta commented:
Yes, it is unrelated to this PR in particular, I will drop this fix so we can move forward. I think it is a good idea to try a global fix in rich text, and I will propose one.

youknowriad reviewed (approved):
LGTM 👍
